jsx/FilterableDataTable.js

```diff
@@ -48,18 +52,48 @@ class FilterableDataTable extends Component {
     history.replaceState({}, '', '?');
   }
 
+  /**
+   * Returns the filter state, with filters that are
+   * set to an invalid option removed from the returned
+   * filters
+   *
+   * @return {object}
+   */
+  validFilters() {
+      let filters = {};
+      this.props.fields.forEach((field) => {
+        const filtername = field.filter.name;
+        const filterval = this.state.filter[filtername];
+        if (!this.state.filter[filtername]) {
+            return;
+        }
+
+        if (field.filter.type !== 'select') {
```

Comment:
I'm not sure how it applies to multiselect since it's not a case that's currently used. As it stands, this PR won't affect any existing tables because none of them have interdependent data. I think the logic of the checking validity becomes much more convoluted (ie cases like what happens if some options are still valid and some aren't?) for multiselects.

I think it might be better to leave this PR only handling simple selects and if the need comes up to have interdependent multiselects handled, handle that when we need to than putting a lot of effort into it (with possibly incorrect assumptions) now.